### PR TITLE
ten-useful-functions

### PR DESCRIPTION
**Pull request by crodeheaver**
Work in progress

### CODE REVIEW

File: camelcase.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+"use strict";
+var sentenceCase = require("./sentenceCase");
+module.exports = function(string){
+    var output = string.match(/\b[^\s]+\b/gi).map(function(value){
```
Comment (barberboy) [inline camelcase.js:4]: You may be aware already but `String.prototype.replace` can take a [function as the replacement value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#Specifying_a_function_as_a_parameter). It calls the function for each match, and use the function's return value as the new value, very similar to what you are doing here.

File: main.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+var base = "A walk to the moon and back.";
+
+var sentence = sentenceCase(base);
+
+var camel = camelcase(base);
+
+var snake = snake_case(camel);
+
+var kebob = kebobcase(camel);
```
Comment (barberboy) [inline main.js:33]: The basic calls look great, but please add some examples of using them with array methods. [.filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Filter) is a natural fit for the greaterthan/lessthan functions, [.reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce) for `largest`, and [.map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Map) for the text.

**Comment by crodeheaver** [comment]
Do these work? @barberboy

**Comment by barberboy** [comment]
:+1:  :100:

**Pull request closed without merging**